apollo-router/src/configuration/mod.rs

```diff
@@ -460,35 +461,88 @@ pub fn validate_configuration(raw_yaml: &str) -> Result<(), ConfigurationError>
                     .enumerate()
                     .filter_map(|(idx, e)| {
                         if let Some(element) = yaml.get_element(&e.instance_path) {
-                            // Dirty hack.
-                            // If the element is a string and it has env variable expansion then we can't validate
-                            // Leave it up to serde to catch these.
-                            if let yaml::Value::String(value, _) = element {
-                                if &envmnt::expand(
-                                    value,
-                                    Some(ExpandOptions::new().clone_with_expansion_type(
-                                        ExpansionType::UnixBracketsWithDefaults,
-                                    )),
-                                ) != value
-                                {
-                                    return None;
+                            match element {
+                                yaml::Value::String(value, marker) => {
+                                    // Dirty hack.
+                                    // If the element is a string and it has env variable expansion then we can't validate
+                                    // Leave it up to serde to catch these.
+                                    if &envmnt::expand(
+                                        value,
+                                        Some(ExpandOptions::new().clone_with_expansion_type(
+                                            ExpansionType::UnixBracketsWithDefaults,
+                                        )),
+                                    ) != value
+                                    {
+                                        return None;
+                                    }
+
+                                    let lines = yaml_split_by_lines
+                                        [0.max(marker.line() as isize - 5) as usize..marker.line()]
+                                        .iter()
+                                        .join("\n");
+
+                                    Some(format!(
+                                        "{}. {}\n\n{}\n{}^----- {}",
+                                        idx + 1,
+                                        e.instance_path,
+                                        lines,
+                                        " ".repeat(0.max(marker.col())),
+                                        e
+                                    ))
+                                }
+                                seq_element @ yaml::Value::Sequence(_, m) => {
+                                    let (start_marker, end_marker) = (m, seq_element.end_marker());
+
+                                    let offset = 0.max(start_marker.line() as isize - 5) as usize;
```

Comment:
NIT: Make 5 a constant with a good name in all places where it is used.

also, wouldn't this be cleaner as
```
let offset = 5.max(start_marker.line()) - 5;
```
? (still using the well named CONSTANT)